File: part1-coding/03-bfs-level-order/python/03_bfs_level_order.py
from collections import deque
class Node:
    def __init__(self,val,left=None,right=None):
        self.val=val; self.left=left; self.right=right
def level_order(root):
    if not root: return []
    q,res=deque([root]),[]
    while q:
        lvl=[]
        for _ in range(len(q)):
            n=q.popleft(); lvl.append(n.val)
            if n.left: q.append(n.left)
            if n.right: q.append(n.right)
        res.append(lvl)
    return res

File: part1-coding/03-bfs-level-order/cpp/03-bfs-level-order.cpp
#include <bits/stdc++.h>
using namespace std;
struct Node{ int val; Node*left; Node*right; Node(int v):val(v),left(NULL),right(NULL){} };
vector<vector<int>> levelOrder(Node* root){
    vector<vector<int>> res; if(!root) return res;
    queue<Node*> q; q.push(root);
    while(!q.empty()){
        int sz=q.size(); vector<int> lvl;
        for(int i=0;i<sz;++i){
            Node* n=q.front(); q.pop(); lvl.push_back(n->val);
            if(n->left) q.push(n->left);
            if(n->right) q.push(n->right);
        } res.push_back(lvl);
    } return res;
}
int main(){ return 0; }
